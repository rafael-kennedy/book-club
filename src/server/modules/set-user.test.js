const express = require("express");
const supertest = require("supertest");
const setUser = require("./set-user");
const tokens = require("../../test/utils/fixture-tokens");
let app;
let request;

describe("set user method", () => {
  beforeAll(async () => {
    app = express();
    app.post("/test", setUser);
    request = supertest(app);
  });

  test("throws 403 on missing token", async () => {
    const res = await request
      .post("/test")
      .send({ payload: true })
      .expect(403);

    expect(res.body).toMatchSnapshot();
  });

  test("throws 403 on expired token", async () => {
    const res = await request
      .post("/test")
      .set("Authorization", "Bearer " + tokens.oldToken)
      .send({ payload: true })
      .expect(403);

    expect(res.body).toMatchSnapshot();
  });

  test("throws 403 on incorrectly signed token", async () => {
    const res = await request
      .post("/test")
      .set("Authorization", "Bearer " + tokens.invalidToken)
      .send({ payload: true })
      .expect(403);

    expect(res.body).toMatchSnapshot();
  });

  test("sets user on correctly signed token", async () => {
    app.post("/test", (req, res) => {
      expect(req.user).toMatchSnapshot();
      res.status(200).send("ok");
    });

    const res = await request
      .post("/test")
      .set("Authorization", "Bearer " + tokens.validToken)
      .send({ payload: true })
      .expect(200);

    expect(res.body).toMatchSnapshot();
  });
});
