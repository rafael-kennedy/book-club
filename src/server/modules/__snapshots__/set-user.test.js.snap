// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set user method sets user on correctly signed token 1`] = `
Object {
  "_id": "5c05e8a02cfef52e5b197fb5",
  "email": "test@test.com",
}
`;

exports[`set user method sets user on correctly signed token 2`] = `Object {}`;

exports[`set user method throws 403 on expired token 1`] = `
Object {
  "message": "Your token has expired. Please log in again",
}
`;

exports[`set user method throws 403 on incorrectly signed token 1`] = `
Object {
  "message": "Your token is invalid. Please log in again",
}
`;

exports[`set user method throws 403 on missing token 1`] = `
Object {
  "message": "You must include your bearer token in the Authorization header of the request",
}
`;
